plonk-core/src/proof_system/proof.rs

```diff
@@ -46,73 +43,80 @@ use rand_core::OsRng;
 /// construct the Proof.
 #[derive(CanonicalDeserialize, CanonicalSerialize, derivative::Derivative)]
 #[derivative(
-    Clone(bound = ""),
-    Debug(bound = ""),
-    Default(bound = ""),
-    Eq(bound = ""),
-    PartialEq(bound = "")
+    Clone(bound = "PC::Commitment: Clone, PC::Proof : Clone"),
+    Debug(
+        bound = "PC::Commitment: std::fmt::Debug, PC::Proof : std::fmt::Debug"
+    ),
+    Default(bound = "PC::Commitment: Default, PC::Proof : Default"),
+    Eq(bound = "PC::Commitment: Eq, PC::Proof : Eq"),
+    PartialEq(bound = "PC::Commitment: PartialEq, PC::Proof : PartialEq")
 )]
-pub struct Proof<E, P>
+pub struct Proof<F, PC>
 where
-    E: PairingEngine,
-    P: TEModelParameters<BaseField = E::Fr>,
+    F: PrimeField,
+    PC: HomomorphicCommitment<F>,
 {
     /// Commitment to the witness polynomial for the left wires.
-    pub(crate) a_comm: Commitment<E>,
+    pub(crate) a_comm: PC::Commitment,
 
     /// Commitment to the witness polynomial for the right wires.
-    pub(crate) b_comm: Commitment<E>,
+    pub(crate) b_comm: PC::Commitment,
 
     /// Commitment to the witness polynomial for the output wires.
-    pub(crate) c_comm: Commitment<E>,
+    pub(crate) c_comm: PC::Commitment,
 
     /// Commitment to the witness polynomial for the fourth wires.
-    pub(crate) d_comm: Commitment<E>,
+    pub(crate) d_comm: PC::Commitment,
 
     /// Commitment to the permutation polynomial.
-    pub(crate) z_comm: Commitment<E>,
+    pub(crate) z_comm: PC::Commitment,
 
     /// Commitment to the quotient polynomial.
-    pub(crate) t_1_comm: Commitment<E>,
+    pub(crate) t_1_comm: PC::Commitment,
 
     /// Commitment to the quotient polynomial.
-    pub(crate) t_2_comm: Commitment<E>,
+    pub(crate) t_2_comm: PC::Commitment,
 
     /// Commitment to the quotient polynomial.
-    pub(crate) t_3_comm: Commitment<E>,
+    pub(crate) t_3_comm: PC::Commitment,
 
     /// Commitment to the quotient polynomial.
-    pub(crate) t_4_comm: Commitment<E>,
+    pub(crate) t_4_comm: PC::Commitment,
+
+    /// Batch opening of the commitments
+    pub aw_opening: PC::Proof,
+
+    /// Batch openings of the shifted commitments
+    pub saw_opening: PC::Proof,
```

Comment:
Can we add in the comments explicitly that is the `aggregated_witness` opening?

I think instead of `commitments` you meant `witnesses`

Reply:
Ok, I think I changed it to what you suggested?

Reply:
Yep!!! Thanks!!